### ISSUE TITLE
Big Key Doesn't Appear in Start Menu

### ISSUE THREAD

**Issue by BlueViper85** [Dessyreqt/alttprandomizer#48]
When in a dungeon the big key does not appear in the start menu inventory anywhere.  This was in V5, seed C9716193

**Comment by KatDevsGames** [comment]
I have a pretty good guess why this is happening. This should be fixed in the update in 2wks.

**Comment by KatDevsGames** [comment]
Requesting more detailed steps to reproduce. Currently working for me.

**Comment by BlueViper85** [comment]
Steps to recreate are uncertain right now.  I'll have to play some more and see how it goes.  However; here are several videos showing this happening.

Ice Palace:
Here's me getting the big key: https://www.twitch.tv/blueviper85/v/81103833?t=02h14m51s

Here's a few seconds later where I menu to switch to the bombs, but don't have the big key listed:
https://www.twitch.tv/blueviper85/v/81103833?t=02h15m12s

Turtle Rock:

Video of me getting the Big Key: https://www.twitch.tv/blueviper85/v/81103833?t=02h34m57s

I did open the usual big key chest later, and it didn't show up after that, so that chest seems to have no bearing on it.

Here is me opening the Big Chest, with a menu just before showing no big key: https://www.twitch.tv/blueviper85/v/81103833?t=02h39m33s

Here's a video showing me missing the Big Key in Turtle Rock, but going through the Big Key door at the end to beat Trinexx: https://www.twitch.tv/blueviper85/v/81103833?t=02h43m36s

**Comment by BlueViper85** [comment]
Just tried with a different seed I already had states loaded, and I couldn't recreate.  While I don't think this is actually seed-specific, I don't have a clue what specific circumstances would create this sort of situation.

I'll see if I still have the ROM for this and if so, load up my old file where this happened and go through some of the dungeons and see if it's all of them, or just some of them (or find out if they return after coming back?).

**Comment by BlueViper85** [comment]
Seed was C9716193.  I have an SRM file if necessary.  Loaded up the original save, and found I have no keys in my pause menu in any of the following dungeons:

-Turtle Rock
-Ice Palace
-Thieves' Town
-Palace of Darkness
-Eastern Palace

Looking through my VoD I found that in Eastern I had the big key in my start menu after opening the big key chest.  I couldn't continue due to lack of bow, so I left and wound up beating Agahnim next.  Then I went to PoD and couldn't get the big key after temporarily bow-locking myself (big key behind a door I couldn't get a key for until I found the bow).  Mirrored, walked out, and Saved and Quit.  Went to Thieves' Town, watched myself pickup the big key, then the next mirror after that (a few minutes later) the Big Key did not appear.  This suggests it may have resulted from completing Agahnim early, at least for now.  Nothing else seems too out of the ordinary.  I didn't visit any great fairies or anything.

**Comment by BlueViper85** [comment]
So I started a new save (slot 2) on the same seed, and went through Escape, went through and grabbed big key in Eastern, made sure it was in the menu, then SQd.  After I reloaded, I returned, and the key was still in the menu, so I went and beat Agahnim.  After beating Agahnim and getting through the text on the Pyramid I SQd and immediately went to Eastern where I checked my start menu and...........   the key was there.

So I guess this rules out beating Agahnim in a similar context as being the root of the issue.  I'll have to revisit my VoD again and see what happens.

I remembered that this was a race with DarkMagician1184 so I checked his VoD very briefly as well, and he experienced it as well.

Here's a clip of him getting the key in Turtle Rock: https://www.twitch.tv/darkmagician1184/v/81116204?t=34m15s

About 20 seconds later you see his menu doesn't have the big key in his menu (Link: https://www.twitch.tv/darkmagician1184/v/81116204?t=34m30s, starts at 30 seconds because any time after seems to cause a black screen and can cause the pause menu to be missed)

But while watching that VoD I had another idea, I think I might know what happened, so I'm going back to testing...

**Comment by BlueViper85** [comment]
Found the issue.  Reloaded the game, went to Maze Race and got the mitts, then went and got the Pegasus Boots. Then I went to Thieve's Town and got the big key, and left.  Returned, and validated that the big key showed in the menu, and it did.

Left the dungeon, went to the Chest Game, opened the chest at the bottom of the far-left column, then went to the chest at the bottom of the column second from the left and got Red Mail.  Went back to Thieves' Town and the Big Key no longer appears in the pause menu.  Went to Eastern Palace and the key was gone from the pause menu there as well.

To help confirm it was getting the item from the Chest Game, I reloaded back to Thieves' Town with the key appearing, left and went to the Chest Game.  I spoke with the guy, then left without opening any chests and went back to Thieves' Town and the key was in the menu.  I left again and went back, spoke with the Chest Game guy and opened a single chest and went back, and again the key was in the menu.  Went back a third time and opened both chests, (getting the Red Mail again) and when I returned to Thieves' Town the key was gone from the menu.

The only remaining question I have right now is whether or not it's the game in general, or only specific items from that game (possibly even specifically just the Red Mail appearing in the Chest Game)

**Comment by BlueViper85** [comment]
Three final notes because I was incredibly curious and wanted to test this as far as I could:
1. This affects both the Big Key and the Big Key with the Chest behind it graphics in the menu (possibly an obvious statement, but worth making)
2. I loaded up another Seed (Specifically C6013115) and ultimately got: Blue Mail, Cane of Byrna, One Bottle, the shovel, the net, the Hammer, Mitts, Moon Pearl, Flippers, and Boots and then went to Thieves' Town and got the big key.  After mirroring out, leaving and re-entering the key was still in my menu.  So I went to the Chest Game and got the item there (in this case the Quake Medallion) and went back to Thieves' Town and found that the Big Key still appeared in the start menu.  Since Pegasus Boots were in Eastern's Big Key chest I went back there and the graphic still appeared in the pause menu.  Strong evidence that this is specific to the Red Mail from the chest game.
3. Given that this involves the red mail, I took it a little further and gathered the items necessary to get Blue Mail in the original seed (C9716193).  After obtaining it I went back to Thieves' Town and validated that the key still appeared in the pause menu, which it did.  I then went and got Red Mail from the Chest Game and then checked Thieves' Town and found that the key with chest graphic was missing still.  So that rules out possibly going from Green Mail straight to Red Mail from the Chest Game.  I also made sure to use completely different chests, just to be sure that the specific chest that gave the item didn't somehow make a difference (it did not)

Remaining questions I'm not equipped well-enough to answer right now:
1. Does the Red Mail have to be in the Chest Game for this happen, or does it happen if it's in other chests?
2. Do other items cause this same issue?

**Comment by ChristosOwen** [comment]
I imagine the issue is rather that Red Mail is a Big Chest item from Ganon's Tower. Finding another dungeon's Big Chest item after already finding the Big Key of a dungeon may corrupt it? Not having a dungeon's usual Big Chest item but having the Big Key might corrupt it (and vice versa)?

**Comment by BlueViper85** [comment]
I think I get what you're saying. But if that were the case, wouldn't this happen with red mail from other chests?  As far as I know we've had seeds that had Red Mail but the keys still appeared.

I'm going to test a little further today. Specifically whether or not Red Mail from other chests causes it and wha happens if I open the big chest in Ganon's Tower since I never opened that one on the original seed.  I'll also see about finding a dungeon I have the key for not the big chest item and also haven't opened the big chest.

**Comment by BlueViper85** [comment]
I went back through an already-completed v5 seed C4496861.  Red Mail was in Ganon's Tower and had already been obtained, I went back through several dungeons and the Big Key was missing from the pause menu in each instance.  The dungeons I checked were:

-Thieves' Town
-Skull Woods (Where I opened the Big Chest, so I know I had the big key)
-Misery Mire
-Turtle Rock
-Ganon's Tower

I went through Ganon's Tower's lower level again and opened the Big Chest which had no impact on whether or not these appeared.

I mirrored out of Ganon's Tower, re-entered and the key was still missing.  I went back to Turtle Rock and it was still missing there as well.

While this doesn't necessarily mean it is the Red Mail since I couldn't go get it (already had it when I beat it) it seems to suggest Red Mail being in the Chest Game is not the specific cause. I'll probably go trough this seed fresh again, see if I can find a v4 check, and also still need to check the newer seeds I was sent to test.

**Comment by BlueViper85** [comment]
TL;DR of the below, this happens immediately upon picking up red mail based on my testing.

Loaded up the same Seed (C9716193) generated by Dessyreqt from a newer version of Randomizer.  This one had the Red Mail back in Ganon's Tower, in the compass room, top-left chest.  I started an all new save and played through the game fairly normally.  By the tim eI got to Ganon's Tower All dungeons have shown the big key up to that point.

Inside Ganon's Tower I made the following notes:
1. When I got the big key, the pause menu shows that I got the big chest, even though I didn't yet and did NOT already have the red mail (screenshot attached)
2. Picked up the red mail, went and got the big key and the big key was not in the pause menu.
3. Went to Turtle Rock and the big key was no longer in the pause menu there either
4. Loaded a save state, picked up the big key then went to the room with the red mail, killed the enemies, grabbed Red Mail, and the Big Key was gone from the pause menu
5. Mirrored out, went to Turtle Rock and the big key was no longer in the pause menu there either

So this appears to be tied directly to red mail as far as I can tell.  Regardless of it's location.

Here's the screenshot from note number 1.  Note the red mail, and this is immediately after picking up the Big Key, I could not have picked up the big chest before this screenshot was taken.

![gtower - big key with chest](https://cloud.githubusercontent.com/assets/12190332/17465228/5faa60c6-5cb7-11e6-8aeb-b6ca6f8b3f8e.png)

**Comment by KatDevsGames** [comment]
Possibly resolved pending further testing.

**Comment by BlueViper85** [comment]
Loaded the ROM you sent me and the big key seems to exist.  In save states I had where the key was already lost, as soon as I got another Dungeon Item the big key appeared.

**Comment by KatDevsGames** [comment]
Yeah. Don't rely 100% on un-fresh savestates but it seems to be working now?

**Comment by BlueViper85** [comment]
Understood.  Yeah it seems fine in fairly limited play testing so far.

**Comment by BlueViper85** [comment]
V6 seems good in this regard.  Big Key existed before Red Mail and after throughout several dungeons.
